scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -209,6 +209,9 @@ object Execution {
     override def join[T, U](t: Execution[T], u: Execution[U]): Execution[(T, U)] = t.zip(u)
   }
 
+  def withConfig[T](ex: => Execution[T])(c: Config => Config): Execution[T] =
```

Comment:
if this is lazy, the arg in the constructor needs to be, otherwise this should not be lazy.